### ISSUE TITLE
Handle promise rejections in main

### ISSUE THREAD

**Issue by sgronblo** [FoalTS/foal#534]
The main function of https://github.com/FoalTS/foal/blob/dd968471881ed7c225b7e72e49bc4f8ed150cef6/packages/examples/src/index.ts#L31 is run without handlers.

This means that an unhandled version is still reported with exit code 0 for current versions of node. There is a message saying: [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code."

This means that running `supervisor -w build --no-restart-on error build/index.js` will keep on restarting in an infinite loop if there is an error in main. Such as "Error: No connection options were found in any of configurations file." which I was getting.

So in order to do this, we should add a handler like below to prevent the infinite loop:

```
main().then(console.log, err => { console.error(err); process.exit(FAILURE_CODE) })
```

**Comment by LoicPoullain** [comment]
Thank you for reporting this @sgronblo! It's true that this infinite loop is really annoying.

New projects will have this new line added 👍

**Comment by LoicPoullain** [comment]
Fixed in v1.1.0
